structure_tests/types.go

```diff
@@ -20,7 +20,7 @@ import (
 )
 
 type StructureTest interface {
-	RunAll(t *testing.T)
+	RunAll(t *testing.T) int
```

Comment:
I thought about doing this. only reason I didn't is because I'm not 100% sure of the behavior of the parser, so I was worried some misconfigured tests could get parsed into structs, then fail later on (maybe because of some bad nested struct or something). this seemed a little more reliable, though definitely not as elegant.